### PR TITLE
OCPQE-24602 fix config issue on 4.17 arm upgrade jobs

### PR DESCRIPTION
**Pull request by jiajliu**
1. Fix the 4.17 upgrade job config issue from https://github.com/openshift/release/pull/54291/files, more details in https://redhat-internal.slack.com/archives/CF8SMALS1/p1723184881747919
2. Also update chain upgrade jobs together for arm arch.

### CODE REVIEW

**Comment by jiajliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-stable-4.17-upgrade-from-stable-4.17-gcp-ipi-compact-filestore-csi-f28

**Comment by jiajliu** [comment]
Original error should pass now in [job](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55388/rehearse-55388-periodic-ci-openshift-openshift-tests-private-release-4.17-arm64-stable-4.17-upgrade-from-stable-4.17-gcp-ipi-compact-filestore-csi-f28/1821816755547279360) 

@jianlinliu @shellyyang1989 PTAL, thanks.

cc @JianLi-RH for the arch issue.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml [hunk 1]
```diff
@@ -83,6 +124,9 @@ tests:
   cron: 35 11 5 * *
   steps:
     cluster_profile: aws-qe
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:arm64-latest
+      OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: release:arm64-target
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml:129]: Chain upgrade does not need `OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE` setting.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml [hunk 2]
```diff
@@ -32,47 +32,88 @@ base_images:
     namespace: ci
     tag: latest
 releases:
-  intermediate412:
+  arm64-intermediate412:
```
[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml:35]: I think we only need to set `latest` and `arm64-latest`, then set the following lines on job config level:
```
    dependencies:
      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:arm64-latest
```

[the remark this answers is not in the document]
Comment (jiajliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml:35]: To be consistent for easily understanding, I also list them together here, the test is still using arm image
```
        name: release:arm64-intermediate412
      - env: RELEASE_IMAGE_TARGET
```
but if you think no need, I'm fine to drop them.

[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml:35]: I prefer to drop them to keep the ci yaml file clean.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml:129]: Oh, my fault, you are right, `OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE` is needed.

Comment (jiajliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml:129]: chain upgrade will drop last hop and use `cucushift-upgrade-toimage` for last hop upgrade, so it still need OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE, right?

https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh#L436

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jiajliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__arm64-nightly-4.17-upgrade-from-stable-4.11.yaml]: np,done

**Comment by jianlinliu** [comment]
cc @liangxia to review.

**Comment by JianLi-RH** [comment]
Thanks @jiajliu , I didn't realize that arm needed both settings

**Pull request merged**